### PR TITLE
feat(services): adjust the translation API to accept full endpoint URL protocol

### PR DESCRIPTION
**Pull request by proeung**
### Related Ticket(s)

DDS Issue: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/7519
HC JIRA Ticket: https://jsw.ibm.com/browse/HC-2290

### Description

- This PR modifies the URL construction within the translation API services to accept HTTPS call defined in the `data-endpoint` prop/attribute.
- This protocol checking logic allows us to define the destination of the endpoint that outputs the same JSON data structure as what's being defined in the `ibm-dotcom-translations` repo, but the data will be dynamically pulled out of Drupal instead of the manual editing process.

### Changelog

**Changed**

-  Adjust the `url` variable within the `fetchTranslation` method.

### Testing
- Pull down the changes of this PR in a test preview/application (see - [link](https://github.com/carbon-design-system/carbon-for-ibm-dotcom-web-components-test/blob/master/src/layouts/cloud-base.hbs#L216)) and define the following markup.
- Ensure that the masthead is rendering the data is from the defined endpoint.
```
<dds-masthead-container
  platform="Security"
  data-endpoint="https://1.www.s81c.com/common/carbon-for-ibm-dotcom/translations/security-masthead"
></dds-masthead-container>
```

### CODE REVIEW

File: packages/services/src/services/Translation/Translation.js [hunk 1]
```diff
@@ -156,9 +156,20 @@ class TranslationAPI {
...
+        // Check to see if the string from the endpoint variable contains https/http or not.
+        if (regex.test(endpoint)) {
+          url = `${endpoint || _ddsEndpoint}/${
+            country !== 'undefined' ? `${country}${lang}` : `${lang}`
+          }.json`;
+        } else {
+          url = `${_host}${endpoint || _ddsEndpoint}/${
+            country !== 'undefined' ? `${country}${lang}` : `${lang}`
+          }.json`;
+        }
```
[the remark this answers is not in the document]
Comment (proeung) [reply packages/services/src/services/Translation/Translation.js]: I like it and a lot cleaner to read than what I previously had.

File: packages/services/src/services/Translation/Translation.js [hunk 2]
```diff
@@ -156,9 +156,20 @@ class TranslationAPI {
...
-        }.json`;
+        let url;
+        //eslint-disable-next-line
+        const regex = /((http(s?))\:\/\/)/g;
+
+        // Check to see if the string from the endpoint variable contains https/http or not.
+        if (regex.test(endpoint)) {
+          url = `${endpoint || _ddsEndpoint}/${
+            country !== 'undefined' ? `${country}${lang}` : `${lang}`
+          }.json`;
```
[the remark this answers is not in the document]
Comment (proeung) [reply packages/services/src/services/Translation/Translation.js]: This is an open question for the business. Right now, both the Global `<dds-masthead-container>` and Cloud `<dds-cloud-masthead-container>` are relying on the Translation API to detect the locale and fetches the correct translation return based on the country code and language code within the URL of the page. This method has proven to be beneficial for the adoption of the Masthead on content ecosystems where there's no mechanism in place to easily input the `cc-lc`.

**Next step:** I think we'll need to touch base with @kjwhitla to see what's the long-term goal of this project elevate navigation is and determine the plan from there, especially related to the end of year launch that we're shooting for.

**Review by emyarod — COMMENTED** [review]
looks good to me! just want to double check, do we want to perform additional validation on the endpoint URL (https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/7537#discussion_r739091150) or is it safe to assume that it'll only handle proper http/https links?

**Comment by proeung** [comment]
@emyarod @annawen1 Just following up on the discussion here (see [comment](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/7537#discussion_r739515029)), @e0ipso and I met with @kjwhitla today and we decided to keep using the `ibm-dotcom-translations` repo as a place to manage the Masthead data for the Storage & QRadar products in order to meet the early December Drupal deployment deadline and keep the custom API endpoint implementation to be something that we revisit at a different time.

With that being said, I think the changes here are still helpful to have when AEM is at a place where the data storage for managing the Masthead content is ready and there's a need to define a different endpoint host (checking again`http/https`) other than the `1.www.s81c.com` host URL. I think we can go ahead and merge this PR down if there are no changes needed, but let me know what you guys think.

cc: @jeffchew

**Review by emyarod — APPROVED** [review]

**Pull request merged**